meta-resin-common/recipes-support/resin-device-register/resin-device-register/resin-device-register

```diff
@@ -44,25 +46,28 @@ fi
 while true; do
     if curl -sL -w '%{http_code}' $API_ENDPOINT/ping -o /dev/null | egrep "20[0-9]|30[0-9]" >/dev/null
     then
-        read api_key user_id application_id device_type uuid <<<$(jq -r '.apiKey,.userId,.applicationId,.deviceType // "raspberry-pi",.uuid' $CONFIG_PATH)
+        read api_key user_id application_id device_type registered_at <<<$(jq -r '.apiKey,.userId,.applicationId,.deviceType // "raspberry-pi",.registered_at // empty' $CONFIG_PATH)
+
+        if [ -n "${registered_at}" ]; then
+            echo "[INFO] resin-device-register : Device already registered."
+            exit 0
+        fi
 
         status_code=$(curl -s -X POST -w "%{http_code}" \
-                    -o /var/log/register-device-curl.log \
-                    -d "user=$user_id" \
-                    -d "application=$application_id" \
-                    -d "uuid=$uuid"  \
+                    --output $LOG_FILE \
+                    --data-urlencode "user=$user_id" \
+                    --data-urlencode "application=$application_id" \
                     --data-urlencode "device_type=$device_type" \
-                    "$API_ENDPOINT/ewa/device?apikey=$api_key")
+                    "$API_ENDPOINT/device/register?apikey=$api_key")
 
         if [ "$status_code" -eq 201 ]; then
-            device_id=$(jq -r '.id' /var/log/register-device-curl.log)
+            device_id=$(jq -r '.id' $LOG_FILE)
+            uuid=$(jq -r '.uuid' $LOG_FILE)
+            api_key=$(jq -r '.api_key' $LOG_FILE)
             config_json=`cat $CONFIG_PATH`
```

Comment:
OK

On Sat, Aug 6, 2016 at 2:22 AM, Page- <notifications@github.com> wrote:

> In meta-resin-common/recipes-support/resin-device-register/
> resin-device-register/resin-device-register
> <https://github.com/resin-os/meta-resin/pull/223#discussion_r73772542>:
>
> >
> >          if [ "$status_code" -eq 201 ]; then
> > -            device_id=$(jq -r '.id' /var/log/register-device-curl.log)
> > +            device_id=$(jq -r '.id' $LOG_FILE)
> > +            uuid=$(jq -r '.uuid' $LOG_FILE)
> > +            api_key=$(jq -r '.api_key' $LOG_FILE)
> >              config_json=`cat $CONFIG_PATH`
>
> That sqlitedb is only used by the supervisor internally (mainly for other
> things), it'll pick up the initial values from config.json (this is when
> the host os provisions before the supervisor starts), and it still has to
> be written to the config.json as it's used by other things on the host os
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/resin-os/meta-resin/pull/223/files/79aa277877d9d18790b8924f62fc35ca2fc7101b#r73772542>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgGCDt1ePvJRfM2IqVTMBr3Gx5llUUmks5qc8XAgaJpZM4JXve1>
> .
>



-- 
Eugene Mirotin
Senior Frontend Engineer
site: Resin.io <https://resin.io/>, twitter: @resin_io
<https://twitter.com/resin_io>